File: app/src/main/java/in/myinnos/googlecodescanner/MainActivity.java
package in.myinnos.googlecodescanner;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.mlkit.vision.barcode.common.Barcode;
import com.google.mlkit.vision.codescanner.GmsBarcodeScanner;
import com.google.mlkit.vision.codescanner.GmsBarcodeScannerOptions;
import com.google.mlkit.vision.codescanner.GmsBarcodeScanning;

import in.myinnos.googlecodescanner.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @SuppressLint("ClickableViewAccessibility")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.fullscreenContent.setOnClickListener(view -> {

            GmsBarcodeScannerOptions options = new GmsBarcodeScannerOptions.Builder()
                    .setBarcodeFormats(
                            Barcode.FORMAT_QR_CODE,
                            Barcode.FORMAT_AZTEC)
                    .build();

            //GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(this, options);
            GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(this);
            scanner.startScan()
                    .addOnSuccessListener(
                            barcode -> Toast.makeText(getApplicationContext(),
                                    "SUCCESS: " + barcode.getRawValue(), Toast.LENGTH_LONG).show())
                    .addOnFailureListener(
                            e -> Log.d("CODE_SCAN_FAILED", e.getMessage()));
        });
    }
}

File: kotlin/app/src/main/java/in/myinnos/codescanner/MainActivity.kt
package `in`.myinnos.codescanner

import `in`.myinnos.codescanner.databinding.ActivityMainBinding
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.mlkit.vision.barcode.common.Barcode
import com.google.mlkit.vision.codescanner.GmsBarcodeScannerOptions
import com.google.mlkit.vision.codescanner.GmsBarcodeScanning

class MainActivity : AppCompatActivity() {

    var binding: ActivityMainBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding!!.root)

        binding!!.fullscreenContent.setOnClickListener {
            val options = GmsBarcodeScannerOptions.Builder()
                .setBarcodeFormats(
                    Barcode.FORMAT_QR_CODE,
                    Barcode.FORMAT_AZTEC
                )
                .build()

            //GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(this, options);
            val scanner = GmsBarcodeScanning.getClient(this)
            scanner.startScan()
                .addOnSuccessListener { barcode: Barcode ->
                    Toast.makeText(
                        applicationContext,
                        "SUCCESS: " + barcode.rawValue, Toast.LENGTH_LONG
                    ).show()
                }
                .addOnFailureListener { e: Exception ->
                    Log.d(
                        "CODE_SCAN_FAILED",
                        e.message!!
                    )
                }
        }

    }
}